engine/src/flutter/lib/ui/semantics.dart

```diff
@@ -962,6 +1025,318 @@ class SemanticsFlag {
+    this.hasImplicitScrolling = false,
+    this.isMultiline = false,
+    this.isReadOnly = false,
+    this.isFocusable = false,
+    this.isLink = false,
+    this.isSlider = false,
+    this.isKeyboardKey = false,
+    this.isCheckStateMixed = false,
+    this.hasExpandedState = false,
+    this.isExpanded = false,
+    this.hasSelectedState = false,
+    this.hasRequiredState = false,
+    this.isRequired = false,
+  });
+
+  /// SemanticsFlags with everything default to false.
+  static const SemanticsFlags none = SemanticsFlags();
+
+  /// {@macro dart.ui.semantics.hasCheckedState}
+  final bool hasCheckedState;
+
+  /// {@macro dart.ui.semantics.isChecked}
+  final bool isChecked;
+
+  /// {@macro dart.ui.semantics.isSelected}
+  final bool isSelected;
+
+  /// {@macro dart.ui.semantics.isButton}
+  final bool isButton;
+
+  /// {@macro dart.ui.semantics.isTextField}
+  final bool isTextField;
+
+  /// {@macro dart.ui.semantics.isFocused}
+  final bool isFocused;
+
+  /// {@macro dart.ui.semantics.hasEnabledState}
+  final bool hasEnabledState;
+
+  /// {@macro dart.ui.semantics.isEnabled}
+  final bool isEnabled;
+
+  /// {@macro dart.ui.semantics.isInMutuallyExclusiveGroup}
+  final bool isInMutuallyExclusiveGroup;
+
+  /// {@macro dart.ui.semantics.isHeader}
+  final bool isHeader;
+
+  /// {@macro dart.ui.semantics.isObscured}
+  final bool isObscured;
+
+  /// {@macro dart.ui.semantics.scopesRoute}
+  final bool scopesRoute;
+
+  /// {@macro dart.ui.semantics.namesRoute}
+  final bool namesRoute;
+
+  /// {@macro dart.ui.semantics.isHidden}
+  final bool isHidden;
+
+  /// {@macro dart.ui.semantics.isImage}
+  final bool isImage;
+
+  /// {@macro dart.ui.semantics.isLiveRegion}
+  final bool isLiveRegion;
+
+  /// {@macro dart.ui.semantics.hasToggledState}
+  final bool hasToggledState;
+
+  /// {@macro dart.ui.semantics.isToggled}
+  final bool isToggled;
+
+  /// {@macro dart.ui.semantics.hasImplicitScrolling}
+  final bool hasImplicitScrolling;
+
+  /// {@macro dart.ui.semantics.isMultiline}
+  final bool isMultiline;
+
+  /// {@macro dart.ui.semantics.isReadOnly}
+  final bool isReadOnly;
+
+  /// {@macro dart.ui.semantics.isFocusable}
+  final bool isFocusable;
+
+  /// {@macro dart.ui.semantics.isLink}
+  final bool isLink;
+
+  /// {@macro dart.ui.semantics.isSlider}
+  final bool isSlider;
+
+  /// {@macro dart.ui.semantics.isKeyboardKey}
+  final bool isKeyboardKey;
+
+  /// {@macro dart.ui.semantics.isCheckStateMixed}
+  final bool isCheckStateMixed;
+
+  /// {@macro dart.ui.semantics.hasExpandedState}
+  final bool hasExpandedState;
+
+  /// {@macro dart.ui.semantics.isExpanded}
+  final bool isExpanded;
+
+  /// {@macro dart.ui.semantics.hasSelectedState}
+  final bool hasSelectedState;
+
+  /// {@macro dart.ui.semantics.hasRequiredState}
+  final bool hasRequiredState;
+
+  /// {@macro dart.ui.semantics.isRequired}
+  final bool isRequired;
+
+  /// Merges the flags from this object with [other].
```

Comment:
nit: It's unclear what "merges" implies. Consider expanding, e.g. "Combines two sets of flags, such that if a flag it set to true in any of the two sets, the resulting set contains that flag set to true."